superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.jsx

```diff
@@ -462,20 +528,16 @@ class AnnotationLayer extends React.PureComponent {
     }
     if (requiresQuery(sourceType)) {
       return (
-        <SelectControl
+        <AsyncSelect
+          /* key to force re-render on sourceType change */
+          key={sourceType}
           ariaLabel={t('Annotation layer value')}
           name="annotation-layer-value"
-          showHeader
-          hovered
-          description={description}
-          label={label}
+          header={this.renderChartHeader(label, description, value)}
           placeholder=""
```

Comment:
Any reason to disable the default placeholder?

Reply:
No this was just holdover from the SelectControl component I neglected to review, I'll take it out